builtin/game/chatcommands.lua

```diff
@@ -1056,3 +1056,44 @@ core.register_chatcommand("kill", {
 		return handle_kill_command(name, param == "" and name or param)
 	end,
 })
+
+local spawn_spawnpos = minetest.setting_get_pos("static_spawnpoint")
+
+minetest.register_chatcommand("spawn", {
+	params = "",
+	description = "Teleport to the spawn point",
+	func = function(name, param)
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			return false, "Player not found"
+		end
+		if spawn_spawnpos then
+			player:set_pos(spawn_spawnpos)
+			return true, "Teleporting to spawn..."
+		else
+			return false, "The spawn point is not set!"
+		end
+	end,
+})
+
+minetest.register_chatcommand("setspawn", {
+	params = "",
+	description = "Sets the spawn point to your current position",
+	privs = {server = true},
+	func = function(name, param)
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			return false, "Player not found"
+		end
+		local pos = player:get_pos()
+		local x = math.floor(pos.x)
+		local y = math.floor(pos.y)
+		local z = math.floor(pos.z)
```

Comment:
I suggest using `vector.round`. And before you might want to add something like 0.0001 to `pos.y`. If you stand on a full node, your y pos is a full integer + 0.5. `math.floor` would just set the position into the ground.